examples/deployment/new-api.ts

```diff
@@ -22,48 +22,50 @@ async function main() {
       networkOwnerId: "test",
     })
     .createActivityPool("app", {
-      image: "file://golem/node:20",
-      // image: "golem/node:20",
-      // image: "http://golem.io/node:20",
-      // imageHash: "0x30984084039480493840",
-      resources: {
-        minCpu: 4,
-        minMemGib: 8,
-        minStorageGib: 16,
-      },
-      replicas: 10,
-      network: "basic",
-      market: {
-        rentHours: 12 /* REQUIRED */,
-        pricing: {
-          maxStartPrice: 1 /* REQUIRED */,
-          maxCpuPerHourPrice: 1 /* REQUIRED */,
-          maxEnvPerHourPrice: 1 /* REQUIRED */,
+      demand: {
+        image: "file://golem/node:20",
+        // image: "golem/node:20",
+        // image: "http://golem.io/node:20",
+        // imageHash: "0x30984084039480493840",
+        resources: {
+          minCpu: 4,
+          minMemGib: 8,
+          minStorageGib: 16,
+        },
+        market: {
+          rentHours: 12 /* REQUIRED */,
+          pricing: {
+            maxStartPrice: 1 /* REQUIRED */,
+            maxCpuPerHourPrice: 1 /* REQUIRED */,
+            maxEnvPerHourPrice: 1 /* REQUIRED */,
+          },
+          withProviders: ["0x123123"],
+          withoutProviders: ["0x123123"],
+          withOperators: ["0x123123"],
+          withoutOperators: ["0x123123"],
         },
-        withProviders: ["0x123123"],
-        withoutProviders: ["0x123123"],
-        withOperators: ["0x123123"],
-        withoutOperators: ["0x123123"],
       },
-      payment: {},
-      // dataTransferProtocol: "gftp",
+      pool: { replicas: { min: 1 } },
```

Comment:
People also shouldn't be defining parameters of the pool. The `replicas` count should be set at the same level as before the change. It appears that we need a dedicated type for this "ComputePowerDemandSpec".